src/main/java/org/openstreetmap/atlas/checks/commands/FlagStatisticsSubCommand.java

```diff
@@ -0,0 +1,619 @@
+                            String line;
+                            // Read each line (flag) from the log file
+                            while ((line = reader.readLine()) != null)
+                            {
+                                // Parse the json
+                                final JsonObject source = this.gson.fromJson(line,
+                                        JsonObject.class);
+                                // Get the check name
+                                final String checkName = source.get(PROPERTIES).getAsJsonObject()
+                                        .get(GENERATOR).getAsString();
+                                // Add the check to the map
+                                countryCheckMap.get(country).putIfAbsent(checkName, new Counter());
+                                // Increment the counter for the check/country
+                                countryCheckMap.get(country).get(checkName).increment();
+                            }
+                        }
+                    }
+                    catch (final IOException exception)
+                    {
+                        this.outputDelegate.printlnWarnMessage(
+                                String.format("Exception thrown while reading file %s: %s",
+                                        file.getName(), exception.getMessage()));
+                    }
+                });
+
+        return countryCheckMap;
+    }
+
+    /**
+     * Given two maps of counts per check per country, get the difference between the counts.
+     *
+     * @param source
+     *            a 2D {@link Map} of flag {@link Counter}s per check {@link String} per country
+     *            {@link String}
+     * @param target
+     *            a 2D {@link Map} of flag {@link Counter}s per check {@link String} per country
+     *            {@link String}
+     * @return a 2D {@link Map} of flag difference {@link Counter}s per check {@link String} per
+     *         country {@link String}
+     */
+    private Map<String, Map<String, Counter>> getDifference(
+            final Map<String, Map<String, Counter>> source,
+            final Map<String, Map<String, Counter>> target)
+    {
+        final Map<String, Map<String, Counter>> difference = new HashMap<>();
+
+        // For each counter in the source map, add a value to the difference map equal to the
+        // mathematical difference of the companion target counter (defaulting to 0 if absent) and
+        // the source counter
+        source.forEach((country, checkCounts) ->
+        {
+            difference.putIfAbsent(country, new HashMap<>());
+            checkCounts
+                    .forEach((check, count) -> difference.get(country).put(check,
+                            new Counter(target.getOrDefault(country, new HashMap<>())
+                                    .getOrDefault(check, new Counter()).getValue()
+                                    - count.getValue())));
```

Comment:
I think this is backwards. If the `target` (or baseline) is 20, and the `source` (or input) is 10, the result should be -10.

Reply:
I think this gets back to terminology again. The input is the source/baseline, the reference is the target. So input/source/baseline should be older than reference/target.

Reply:
Your example shows -2 as the difference between 4 (input) & 2 (reference); However, the new run (input) has 2 additional flags from the comparison, so it should be +2, right?
```csv
Country,Check,Input,Reference,Difference
ABC,Check2,4,2,-2
```

Reply:
The new run is reference.